src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -1255,6 +1257,43 @@ class spell_pri_guardian_spirit : public AuraScript
     }
 };
 
+// 421558 - Heaven's Wrath
+class spell_pri_heavens_wrath : public AuraScript
+{
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_ULTIMATE_PENITENCE
+        });
+    }
+
+    bool CheckProc(ProcEventInfo& eventInfo)
+    {
+        if (eventInfo.GetSpellInfo()->Id == SPELL_PRIEST_ULTIMATE_PENITENCE)
```

Comment:
This needs to exclude spells 421543 (damage bolt) and 421544 (healing bolt), not 421453 (the aura with cooldown)

Reply:
oh, i see, makes sense